contracts/programs/bubblegum/src/lib.rs

```diff
@@ -321,6 +375,132 @@ pub fn get_instruction_type(full_bytes: &[u8]) -> InstructionName {
     }
 }
 
+fn assert_enough_mints_to_approve(
+    total_mint_capacity: u64,
+    num_mints_approved: u64,
+    num_minted: u64,
+    to_approve: u64,
+) -> Result<()> {
+    let remaining_mints_to_approve = total_mint_capacity.saturating_sub(num_mints_approved);
+    let remaining_mints = total_mint_capacity.saturating_sub(num_minted);
+    msg!(&format!(
+        "{} {} {}",
+        remaining_mints_to_approve, remaining_mints, to_approve
+    ));
+    if to_approve > remaining_mints_to_approve || to_approve > remaining_mints {
+        return Err(BubblegumError::NotEnoughMintCapacity.into());
+    }
+    Ok(())
+}
+
+fn process_mint_request<'info>(
+    mint_authority_request_account: &mut AccountInfo<'info>,
+    merkle_slab: &AccountInfo<'info>,
+    mint_authority: &Signer<'info>,
+    program_id: &Pubkey,
+) -> Result<()> {
+    // Address check
+    let (expected_request_key, _bump) = Pubkey::find_program_address(
+        &[merkle_slab.key().as_ref(), mint_authority.key.as_ref()],
+        program_id,
+    );
+    if expected_request_key != mint_authority_request_account.key() {
+        return Err(BubblegumError::MintRequestKeyMismatch.into());
+    }
+
+    // Data check
+    let mut raw_data = mint_authority_request_account.try_borrow_mut_data()?;
+    let (discriminator, mut data_bytes) = raw_data.split_at_mut(8);
+    let expected_discriminant = MintRequest::discriminator();
+    // - check discriminator
+    if discriminator != expected_discriminant {
+        return Err(BubblegumError::MintRequestDiscriminatorMismatch.into());
+    }
+
+    // - check request fields
+    let mut request = MintRequest::try_from_slice(&data_bytes)?;
+    if request.approved != 1 {
```

Comment:
I think this should be an enum or it should be a member function

Reply:
addressed